python-package/xgboost/core.py

```diff
@@ -15,7 +15,10 @@
 import re
 import sys
 
+import cudf
```

Comment:
See https://github.com/dmlc/xgboost/blob/master/python-package/xgboost/compat.py for how to optionally import these dependencies.

Reply:
Implemented a first pass at this. Would appreciate further review/feedback.